x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -0,0 +1,193 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import { ElasticsearchClient, SavedObjectsClientContract } from '@kbn/core/server';
+import {
+  AgentPolicyServiceInterface,
+  AgentService,
+  PackagePolicyServiceInterface,
+  PackageService,
+} from '@kbn/fleet-plugin/server';
+import { Installation, ListResult, PackagePolicy } from '@kbn/fleet-plugin/common';
+import {
+  CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+  INFO_ROUTE_PATH,
+  LATEST_FINDINGS_INDEX_DEFAULT_NS,
+} from '../../../common/constants';
+import { CspAppContext } from '../../plugin';
+import type { CspRouter } from '../../types';
+import { CspSetupStatus, Status } from '../../../common/types';
+import {
+  addRunningAgentToAgentPolicy,
+  getCspAgentPolicies,
+  getCspPackagePolicies,
+} from '../../lib/fleet_util';
+
+export const INDEX_TIMEOUT_IN_HOURS = 5;
+
+const isFindingsExists = async (esClient: ElasticsearchClient): Promise<boolean> => {
+  try {
+    const queryResult = await esClient.search({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+      query: {
+        match_all: {},
+      },
+      size: 1,
+    });
+
+    const hasLatestFinding = !!queryResult.hits.hits.length;
+
+    return hasLatestFinding ? true : false;
+  } catch (e) {
+    return false;
+  }
+};
+
+const getHealthyAgents = async (
+  soClient: SavedObjectsClientContract,
+  cspPackagePolicies: ListResult<PackagePolicy>,
+  agentPolicyService: AgentPolicyServiceInterface,
+  agentService: AgentService
+): Promise<number> => {
+  const agentPolicies = await getCspAgentPolicies(
+    soClient,
+    cspPackagePolicies.items,
+    agentPolicyService
+  );
+  const enrichAgentPolicies = await addRunningAgentToAgentPolicy(agentService, agentPolicies);
+  const initialValue = 0;
+  const totalAgents = enrichAgentPolicies
+    .map((agentPolicy) => (agentPolicy.agents ? agentPolicy.agents : 0))
+    .reduce((previousValue, currentValue) => previousValue + currentValue, initialValue);
+
+  return totalAgents;
+};
+
+const getTimeDeltaSinceInstallation = (packageInfo: Installation): number => {
+  const installTime = packageInfo!.install_started_at;
+
+  // calculate time delta and convert to hours
+  const dt = (new Date().getTime() - new Date(installTime).getTime()) / (1000 * 60 * 60);
+
+  return dt;
+};
+
+const getStatus = async (
+  esClient: ElasticsearchClient,
+  installedIntegrations: number,
+  healthyAgents: number,
+  timeDeltaSinceInstallation: number
+): Promise<Status> => {
+  if (await isFindingsExists(esClient)) return 'indexed';
+
+  if (installedIntegrations === 0) return 'not installed';
+
+  if (healthyAgents === 0) return 'not deployed';
+
+  if (timeDeltaSinceInstallation < INDEX_TIMEOUT_IN_HOURS) return 'indexing';
+
+  return 'index timeout';
```

Comment:
just a thought, but - 
is `index_timeout` the only state possible here? 

i'd prefer if we could tell when exactly to return the timeout status 

and if none of these happen, we should throw an error to indicate an invalid state